internal/pkg/manifest/storage.go

```diff
@@ -0,0 +1,108 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package manifest
+
+import (
+	"errors"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"gopkg.in/yaml.v3"
+)
+
+var errUnmarshalEFSOpts = errors.New("unmarshal efs field into string or map")
+
+// StorageConfig is embedded into top level manifest structs when unmarshaling.
+type StorageConfig struct {
+	Storage Storage `yaml:"storage"`
+}
+
+// Storage represents the options for external and native storage.
+type Storage struct {
+	Volumes map[string]Volume `yaml:"volumes"`
+}
+
+// Volume is an abstraction which merges the MountPoint and Volumes concepts from the ECS Task Definition
+type Volume struct {
+	EFS            EFSIDOrConfig `yaml:"efs"`
+	MountPointOpts `yaml:",inline"`
+}
+
+// MountPointOpts is shared between Volumes for the main container and MountPoints for sidecars.
+type MountPointOpts struct {
+	ContainerPath *string `yaml:"path"`
+	ReadOnly      *bool   `yaml:"read_only"`
+}
+
+// MountPoint is used to let sidecars mount volumes defined in `storage`
+type MountPoint struct {
```

Comment:
Maybe we can make the name more explicitly showing it is a field for `sidecars`. What do you think?

Reply:
good idea. done.